src/ops.rs

```diff
@@ -190,42 +225,53 @@ pub fn intersect_packages(installed: Vec<MainRepoPackage>, to_update: &[String])
     installed.into_iter().filter(|p| to_update.contains(&p.name)).collect()
 }
 
-/// Download the list of versions of the specified package from the main [`crates.io`](https://crates.io) registry
-/// using the specified auth token.
-///
-/// Plug into `crate_versions()` to convert to machine-readable form.
-///
-/// # Examples
-///
-/// ```
-/// # use cargo_update::ops::crate_versions_raw;
-/// let raw_versions = crate_versions_raw("checksums");
-/// ```
-pub fn crate_versions_raw(crate_name: &str) -> String {
-    let mut buf = String::new();
-    HttpClient::new()
-        .get(&format!("https://crates.io/api/v1/crates/{}/versions", crate_name))
-        .send()
-        .unwrap()
-        .read_to_string(&mut buf)
-        .unwrap();
-    buf
-}
-
-/// Parse the raw crate version list into a collection of `Semver`s.
+/// Parse the raw crate descriptor from the repository into a collection of `Semver`s.
 ///
 /// # Examples
 ///
 /// ```
-/// # use cargo_update::ops::{crate_versions_raw, crate_versions};
-/// # let raw_versions = crate_versions_raw("checksums");
-/// let versions = crate_versions(&raw_versions);
+/// # use cargo_update::ops::crate_versions;
+/// # use std::path::PathBuf;
+/// # let desc_path = PathBuf::from("test-data/checksums-versions.json");
+/// let versions = crate_versions(&desc_path);
 ///
 /// println!("Released versions of checksums:");
 /// for ver in &versions {
 ///     println!("  {}", ver);
 /// }
 /// ```
-pub fn crate_versions(raw: &str) -> Vec<Semver> {
-    json::parse(raw).unwrap()["versions"].members().map(|v| Semver::parse(v["num"].as_str().unwrap()).unwrap()).collect()
+pub fn crate_versions(package_desc: &Path) -> Vec<Semver> {
+    let mut buf = String::new();
+    File::open(package_desc).unwrap().read_to_string(&mut buf).unwrap();
+
+    buf.lines().map(|ver| Semver::parse(json::parse(ver).unwrap()["vers"].as_str().unwrap()).unwrap()).rev().collect()
```

Comment:
Yes, since versions are listed hi-lo in `crates.io` API but lo-hi in the index so we preserve compatibility there. I'll add a note to the docs.

Will do.